modules/c++/sys/source/ExecUnix.cpp

```diff
@@ -24,9 +24,115 @@
 #if !defined(WIN32)
 
 #include <sys/wait.h>
-#include "sys/Exec.h"
+#include <str/Manip.h>
+#include <sys/Exec.h>
 
-int sys::ExecPipe::closePipe()
+#define READ_PIPE  0
+#define WRITE_PIPE 1
+
+namespace sys
+{
+
+FILE* ExecPipe::openPipe(const std::string& command,
+                         const std::string& type)
+{
+    register FILE* ioFile = NULL;
+    int pIO[2];
+
+    //! create the IO pipes for stdin/out
+    if (pipe(pIO) < 0) { return NULL; }
+
+    //! fork a subprocess for running our command --
+    //  here we use the user-defined pid, which is one major
+    //  differences between this and the normal popen()
+    switch (mProcess = fork())
```

Comment:
For clarity, avoid assignments inside other statements.  So assign `mProcess`, then `switch` on it.